sechub-commons-model/src/main/java/com/mercedesbenz/sechub/commons/model/SecHubConfigurationModelValidator.java

```diff
@@ -284,41 +285,108 @@ private void handleHTTPHeaders(InternalValidationContext context, SecHubWebScanC
             return;
         }
 
+        String targetUrlLowerCased = addTrailingSlashToUrlWhenMissingAndLowerCase(webScan.getUrl().toString());
+
+        // A simple map to keep track of headers an the urls they should be sent to
+        // With this map we validate the header configurations and inform the user about
+        // invalid combinations
+        Map<String, Boolean> occurencesMap = new HashMap<>();
+
         List<HTTPHeaderConfiguration> httpHeaders = optHttpHeaders.get();
         for (HTTPHeaderConfiguration httpHeader : httpHeaders) {
-            if (httpHeader.getName() == null || httpHeader.getName().isEmpty()) {
-                context.result.addError(WEB_SCAN_NO_HEADER_NAME_DEFINED);
-                return;
-            }
-            if (httpHeader.getValue() == null || httpHeader.getValue().isEmpty()) {
-                context.result.addError(WEB_SCAN_NO_HEADER_VALUE_DEFINED);
+            if (!assertHeaderNameAndHeaderValueAreSet(context, httpHeader)) {
                 return;
             }
+
+            String headerNameLowerCased = httpHeader.getName().toLowerCase();
+
             if (!httpHeader.getOnlyForUrls().isPresent()) {
+                // Map key is a combination of the lowercased headerName and target URL
+                // Setting the target URL here means the header is send for every URL
+                // But this way it is easier to check if the configuration is valid
+                String mapKey = headerNameLowerCased + ":" + targetUrlLowerCased;
+                if (occurencesMap.putIfAbsent(mapKey, true) != null) {
+                    context.result.addError(WEB_SCAN_MULTIPLE_DEFAULT_URLS_FOR_THE_SAME_TYPE_OF_HEADER_DEFINED);
+                    return;
+                }
                 continue;
```

Comment:
To remove the continue and the returns simply add an else branch

I would also prefer 

```java
if (httpHeader.getOnlyForUrls().isPresent()) {
    List<String> onlyForUrls = httpHeader.getOnlyForUrls().get();
            validateOnlyForUrls(context, targetUrlLowerCased, occurencesMap, headerNameLowerCased, onlyForUrls));}
else{
    /* no "not only for url" - means this is the default when not explicit defined */
    String mapKey = headerNameLowerCased + ":" + targetUrlLowerCased;
    if (occurencesMap.putIfAbsent(mapKey, true) != null) {
          /* there was already a default setting defined for this target URL - but only one may be defined!*/ 
         context.result.addError(WEB_SCAN_MULTIPLE_DEFAULT_URLS_FOR_THE_SAME_TYPE_OF_HEADER_DEFINED);

    }
}
```

Reply:
Reading it again with the changes, It became clear that there must be 2 different for loops:

1. detect defaults + duplications
2. if there are no duplications check for path duplications

Reason: I am not sure, but currently the validateOnlyForUrls method call uses a map which could be not completely filled at some time